core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -286,15 +286,15 @@ class CassandraJournal(cfg: Config)
         result.map(_ => extractTagWrites(serialized))
     }
 
-    tws.onComplete { result =>
-      self ! WriteFinished(pid, p.future)
-      p.success(Done)
-      // notify TagWriters when write was successful
-      result.foreach(bulkTagWrite => tagWrites.foreach(_ ! bulkTagWrite))
-    }(akka.dispatch.ExecutionContexts.sameThreadExecutionContext)
-
-    //Nil == all good
-    tws.map(_ => Nil)(akka.dispatch.ExecutionContexts.sameThreadExecutionContext)
+    bulkTagWrite.map(btw => {
+      // notify TagWriters when write was successful before completing the future otherwise
+      // we can get another seq of AtomicWrites for the same persistent actor before this is sent
+      tagWrites.foreach(_ ! btw)
+      self ! WriteFinished(pid, writeInProgressForPersistentId.future)
+      writeInProgressForPersistentId.success(Done)
```

Comment:
now these two lines are not done if the bulkTagWrite fails

Reply:
ofc, forgot about that